File: backend/src/main/java/com/video/entity/Video.java
package com.video.entity;

import javax.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "video_metadata")
public class Video {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, unique = true, length = 36)
    private String uuid;
    
    @Column(nullable = false)
    private String title;
    
    @Column(name = "original_filename", length = 500)
    private String originalFilename;
    
    @Column(name = "original_path", length = 1000)
    private String originalPath;

    @Column(name = "source_type", length = 20)
    private String sourceType = "local_file";

    @Column(name = "alist_path", length = 1000)
    private String alistPath;
    
    @Column(name = "hls_path", length = 1000)
    private String hlsPath;
    
    @Column(name = "thumbnail_path", length = 1000)
    private String thumbnailPath;
    
    @Column(name = "duration")
    private Long duration = 0L;
    
    @Column(name = "file_size")
    private Long fileSize = 0L;
    
    @Column
    private Integer width = 0;
    
    @Column
    private Integer height = 0;
    
    @Column(nullable = false, length = 20)
    private String status = "pending";
    
    @Column(name = "current_position")
    private Long currentPosition = 0L;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @Column(name = "category_id")
    private Long categoryId;
    
    @Column(columnDefinition = "TEXT")
    private String overview;
    
    @Column(name = "poster_path", length = 1000)
    private String posterPath;
    
    @Column(name = "backdrop_path", length = 1000)
    private String backdropPath;
    
    @Column(name = "tmdb_id")
    private Long tmdbId;
    
    @Column(name = "imdb_id", length = 20)
    private String imdbId;
    
    @Column(name = "douban_id", length = 20)
    private String doubanId;
    
    @Column
    private Double rating;
    
    @Column(name = "release_year")
    private Integer releaseYear;
    
    @Column(length = 500)
    private String genres;
    
    @Column(columnDefinition = "TEXT")
    private String actors;
    
    @Column(name = "director", length = 200)
    private String director;
    
    @Column(name = "actor_list_json", columnDefinition = "TEXT")
    private String actorListJson;
    
    @Column(name = "scraping_status", length = 20)
    private String scrapingStatus = "pending";
    
    @Column(name = "series_id")
    private Long seriesId;
    
    @Column(name = "season_id")
    private Long seasonId;
    
    @Column(name = "episode_number")
    private Integer episodeNumber;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/api/video.ts
import axios from 'axios'

const api = axios.create({
  baseURL: '/api',
  timeout: 60000,
})

export interface Video {
  uuid: string
  title: string
  originalFilename: string
  originalPath: string
  hlsPath: string
  thumbnailPath: string
  duration: number
  fileSize: number
  width: number
  height: number
  status: string
  currentPosition: number
  createdAt: string
  updatedAt: string
  posterPath?: string
  backdropPath?: string
  overview?: string
  rating?: number
  releaseYear?: number
  genres?: string
  actors?: string
  director?: string
  actorListJson?: string
  seriesId?: number
  seasonId?: number
  episodeNumber?: number
}

export interface VideoUploadResponse {
  uuid: string
  status: string
  message: string
}

export interface Subtitle {
  language: string
  label: string
  status: 'pending' | 'searching' | 'downloaded' | 'failed' | 'not_found'
  fileName?: string
  fileSize?: number
  hasFile: boolean
}

export interface SubtitlesResponse {
  videoUuid: string
  subtitles: Subtitle[]
  total: number
  hasSubtitles: boolean
}

export interface SubtitleStatusResponse {
  videoUuid: string
  hasSubtitles: boolean
  languages: Subtitle[]
}

export const videoApi = {
  upload(file: File, title?: string): Promise<VideoUploadResponse> {
    const formData = new FormData()
    formData.append('file', file)
    if (title) formData.append('title', title)
    return api.post('/videos/upload', formData).then(res => res.data)
  },

  getList(page = 0, size = 20): Promise<{ content: Video[]; totalElements: number }> {
    return api.get('/videos', { params: { page, size } }).then(res => res.data)
  },

  getByUuid(uuid: string): Promise<Video> {
    return api.get(`/videos/${uuid}`).then(res => res.data)
  },

  getStreamUrl(uuid: string): string {
    return `/api/videos/${uuid}/stream`
  },

  updateProgress(uuid: string, position: number): Promise<void> {
    return api.put(`/videos/${uuid}/progress`, { position }).then(res => res.data)
  },

  delete(uuid: string): Promise<void> {
    return api.delete(`/videos/${uuid}`).then(res => res.data)
  },

  getSubtitles(uuid: string): Promise<SubtitlesResponse> {
    return api.get(`/videos/${uuid}/subtitles`).then(res => res.data)
  },

  searchSubtitles(uuid: string): Promise<{ status: string; message: string }> {
    return api.post(`/videos/${uuid}/subtitles/search`).then(res => res.data)
  },

  searchSubtitle(uuid: string, language: string): Promise<{ status: string; message: string }> {
    return api.post(`/videos/${uuid}/subtitles/${language}/search`).then(res => res.data)
  },

  deleteSubtitle(uuid: string, language: string): Promise<{ status: string; message: string }> {
    return api.delete(`/videos/${uuid}/subtitles/${language}`).then(res => res.data)
  },

  getSubtitleStatus(uuid: string): Promise<SubtitleStatusResponse> {
    return api.get(`/videos/${uuid}/subtitles/status`).then(res => res.data)
  },

  getSubtitleUrl(uuid: string, lang: string): string {
    return `/api/subtitles/${uuid}/${lang}.vtt`
  },
}

export default api